File: solution files/Python/part-02/task-02-variable-names/naming_experiment.py
from typing import List


class Product:
    def __init__(self, name: str) -> None:
        self.name = name


def process(d):
    # return the result
    return d


def extract_product_names(products: List[Product]) -> List[str]:
    # iterate over products and return a list of each product's name
    return [product.name for product in products]

File: solution files/C#/part-02/task-02-variable-names/NamingExperiment.cs
using System.Collections.Generic;
using System.Linq;

public class NamingExperiment
{
    public List<string> Process(List<object> d)
    {
        // return the result
        return d.Select(x => x.ToString()).ToList();
    }

    public List<string> ExtractProductNames(List<Product> products)
    {
        // iterate over products and return a list of each product's name
        return products.Select(p => p.Name).ToList();
    }
}

public class Product
{
    public string Name { get; set; } = string.Empty;
}
